Framework/src/TTreeTrend.cxx

```diff
@@ -0,0 +1,167 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file    TTreeTrend.cxx
+/// \author  Piotr Konopka
+///
+
+#include "QualityControl/TTreeTrend.h"
+#include "QualityControl/QcInfoLogger.h"
+#include "QualityControl/DatabaseInterface.h"
+#include "QualityControl/MonitorObject.h"
+#include "QualityControl/Reductor.h"
+#include "RootClassFactory.h"
+#include <Configuration/ConfigurationInterface.h>
+#include <TH1.h>
+#include <TCanvas.h>
+#include <TPaveText.h>
+
+using namespace o2::quality_control;
+using namespace o2::quality_control::core;
+using namespace o2::quality_control::postprocessing;
+
+void TTreeTrend::configure(std::string name, o2::configuration::ConfigurationInterface& config)
+{
+  mConfig = TTreeTrendConfig(name, config);
+}
+
+void TTreeTrend::initialize(Trigger, framework::ServiceRegistry& services)
+{
+  // Preparing data structure of TTree
+  mTrend = std::make_unique<TTree>(); // todo: retrieve last TTree, so we continue trending. maybe do it optionally?
```

Comment:
about your TODO : yes, optionnally but it is an important feature to consider.

Reply:
I treat it as one of the first features that should be added after the initial version is merged.